lib/src/util/flutter_utils.dart

```diff
@@ -90,6 +98,19 @@ class _Flutter {
     return false;
   }
 
+  bool hasWidgetAsAscendant(ClassElement element,
+      [List<ClassElement> alreadySeen]) {
```

Comment:
Consider using a `Set`. I don't know how deep the widget class hierarchy normally gets, so it might not matter, but containment might be faster to compute using a set.